--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/lib/TestAppForComparison/Controller/Root.pm	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/lib/TestAppForComparison/Controller/Root.pm	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,61 @@
+package TestAppForComparison::Controller::Root;
+
+use strict;
+use warnings;
+use parent 'Catalyst::Controller';
+
+#
+# Sets the actions in this controller to be registered with no prefix
+# so they function identically to actions created in MyApp.pm
+#
+__PACKAGE__->config->{namespace} = '';
+
+=head1 NAME
+
+TestAppForComparison::Controller::Root - Root Controller for TestAppForComparison
+
+=head1 DESCRIPTION
+
+[enter your description here]
+
+=head1 METHODS
+
+=cut
+
+=head2 index
+
+=cut
+
+sub index :Path :Args(0) {
+    my ( $self, $c ) = @_;
+
+    # Hello World
+    $c->response->body( $c->welcome_message );
+}
+
+sub default :Path {
+    my ( $self, $c ) = @_;
+    $c->response->body( 'Page not found' );
+    $c->response->status(404);
+}
+
+=head2 end
+
+Attempt to render a view, if needed.
+
+=cut
+
+sub end : ActionClass('RenderView') {}
+
+=head1 AUTHOR
+
+Devin Austin,,,
+
+=head1 LICENSE
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
+
+1;
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/lib/TestAppForComparison.pm	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/lib/TestAppForComparison.pm	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,64 @@
+package TestAppForComparison;
+
+use strict;
+use warnings;
+
+use Catalyst::Runtime 5.80;
+
+# Set flags and add plugins for the application
+#
+#         -Debug: activates the debug mode for very useful log messages
+#   ConfigLoader: will load the configuration from a Config::General file in the
+#                 application's home directory
+# Static::Simple: will serve static files from the application's root
+#                 directory
+
+use parent qw/Catalyst/;
+use Catalyst qw/-Debug
+                ConfigLoader
+                Static::Simple/;
+our $VERSION = '0.01';
+
+# Configure the application.
+#
+# Note that settings in testappforcomparison.conf (or other external
+# configuration file that you set up manually) take precedence
+# over this when using ConfigLoader. Thus configuration
+# details given here can function as a default configuration,
+# with an external configuration file acting as an override for
+# local deployment.
+
+__PACKAGE__->config( name => 'TestAppForComparison' );
+
+# Start the application
+__PACKAGE__->setup();
+
+
+=head1 NAME
+
+TestAppForComparison - Catalyst based application
+
+=head1 SYNOPSIS
+
+    script/testappforcomparison_server.pl
+
+=head1 DESCRIPTION
+
+[enter your description here]
+
+=head1 SEE ALSO
+
+L<TestAppForComparison::Controller::Root>, L<Catalyst>
+
+=head1 AUTHOR
+
+Devin Austin,,,
+
+=head1 LICENSE
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
+
+1;
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/Makefile.PL	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/Makefile.PL	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,20 @@
+#!/usr/bin/env perl
+# IMPORTANT: if you delete this file your app will not work as
+# expected.  You have been warned.
+use inc::Module::Install;
+
+name 'TestAppForComparison';
+all_from 'lib/TestAppForComparison.pm';
+
+requires 'Catalyst::Runtime' => '5.80004';
+requires 'Catalyst::Plugin::ConfigLoader';
+requires 'Catalyst::Plugin::Static::Simple';
+requires 'Catalyst::Action::RenderView';
+requires 'parent';
+requires 'Config::General'; # This should reflect the config file format you've chosen
+                 # See Catalyst::Plugin::ConfigLoader for supported formats
+catalyst;
+
+install_script glob('script/*.pl');
+auto_install;
+WriteAll;
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/README	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/README	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1 @@
+Run script/testappforcomparison_server.pl to test the application.
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/script/testappforcomparison_cgi.pl	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/script/testappforcomparison_cgi.pl	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,37 @@
+#!/usr/bin/env perl
+
+BEGIN { $ENV{CATALYST_ENGINE} ||= 'CGI' }
+
+use strict;
+use warnings;
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use TestAppForComparison;
+
+TestAppForComparison->run;
+
+1;
+
+=head1 NAME
+
+testappforcomparison_cgi.pl - Catalyst CGI
+
+=head1 SYNOPSIS
+
+See L<Catalyst::Manual>
+
+=head1 DESCRIPTION
+
+Run a Catalyst application as a cgi script.
+
+=head1 AUTHORS
+
+Catalyst Contributors, see Catalyst.pm
+
+=head1 COPYRIGHT
+
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/script/testappforcomparison_create.pl	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/script/testappforcomparison_create.pl	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,85 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Getopt::Long;
+use Pod::Usage;
+eval "use Catalyst::Helper;";
+
+if ($@) {
+  die <<END;
+To use the Catalyst development tools including catalyst.pl and the
+generated script/myapp_create.pl you need Catalyst::Helper, which is
+part of the Catalyst-Devel distribution. Please install this via a
+vendor package or by running one of -
+
+  perl -MCPAN -e 'install Catalyst::Devel'
+  perl -MCPANPLUS -e 'install Catalyst::Devel'
+END
+}
+
+my $force = 0;
+my $mech  = 0;
+my $help  = 0;
+
+GetOptions(
+    'nonew|force'    => \$force,
+    'mech|mechanize' => \$mech,
+    'help|?'         => \$help
+ );
+
+pod2usage(1) if ( $help || !$ARGV[0] );
+
+my $helper = Catalyst::Helper->new( { '.newfiles' => !$force, mech => $mech } );
+
+pod2usage(1) unless $helper->mk_component( 'TestAppForComparison', @ARGV );
+
+1;
+
+=head1 NAME
+
+testappforcomparison_create.pl - Create a new Catalyst Component
+
+=head1 SYNOPSIS
+
+testappforcomparison_create.pl [options] model|view|controller name [helper] [options]
+
+ Options:
+   -force        don't create a .new file where a file to be created exists
+   -mechanize    use Test::WWW::Mechanize::Catalyst for tests if available
+   -help         display this help and exits
+
+ Examples:
+   testappforcomparison_create.pl controller My::Controller
+   testappforcomparison_create.pl -mechanize controller My::Controller
+   testappforcomparison_create.pl view My::View
+   testappforcomparison_create.pl view MyView TT
+   testappforcomparison_create.pl view TT TT
+   testappforcomparison_create.pl model My::Model
+   testappforcomparison_create.pl model SomeDB DBIC::Schema MyApp::Schema create=dynamic\
+   dbi:SQLite:/tmp/my.db
+   testappforcomparison_create.pl model AnotherDB DBIC::Schema MyApp::Schema create=static\
+   dbi:Pg:dbname=foo root 4321
+
+ See also:
+   perldoc Catalyst::Manual
+   perldoc Catalyst::Manual::Intro
+
+=head1 DESCRIPTION
+
+Create a new Catalyst Component.
+
+Existing component files are not overwritten.  If any of the component files
+to be created already exist the file will be written with a '.new' suffix.
+This behavior can be suppressed with the C<-force> option.
+
+=head1 AUTHORS
+
+Catalyst Contributors, see Catalyst.pm
+
+=head1 COPYRIGHT
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/script/testappforcomparison_fastcgi.pl	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/script/testappforcomparison_fastcgi.pl	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,79 @@
+#!/usr/bin/env perl
+
+BEGIN { $ENV{CATALYST_ENGINE} ||= 'FastCGI' }
+
+use strict;
+use warnings;
+use Getopt::Long;
+use Pod::Usage;
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use TestAppForComparison;
+
+my $help = 0;
+my ( $listen, $nproc, $pidfile, $manager, $detach, $keep_stderr );
+
+GetOptions(
+    'help|?'      => \$help,
+    'listen|l=s'  => \$listen,
+    'nproc|n=i'   => \$nproc,
+    'pidfile|p=s' => \$pidfile,
+    'manager|M=s' => \$manager,
+    'daemon|d'    => \$detach,
+    'keeperr|e'   => \$keep_stderr,
+);
+
+pod2usage(1) if $help;
+
+TestAppForComparison->run(
+    $listen,
+    {   nproc   => $nproc,
+        pidfile => $pidfile,
+        manager => $manager,
+        detach  => $detach,
+        keep_stderr => $keep_stderr,
+    }
+);
+
+1;
+
+=head1 NAME
+
+testappforcomparison_fastcgi.pl - Catalyst FastCGI
+
+=head1 SYNOPSIS
+
+testappforcomparison_fastcgi.pl [options]
+
+ Options:
+   -? -help      display this help and exits
+   -l -listen    Socket path to listen on
+                 (defaults to standard input)
+                 can be HOST:PORT, :PORT or a
+                 filesystem path
+   -n -nproc     specify number of processes to keep
+                 to serve requests (defaults to 1,
+                 requires -listen)
+   -p -pidfile   specify filename for pid file
+                 (requires -listen)
+   -d -daemon    daemonize (requires -listen)
+   -M -manager   specify alternate process manager
+                 (FCGI::ProcManager sub-class)
+                 or empty string to disable
+   -e -keeperr   send error messages to STDOUT, not
+                 to the webserver
+
+=head1 DESCRIPTION
+
+Run a Catalyst application as fastcgi.
+
+=head1 AUTHORS
+
+Catalyst Contributors, see Catalyst.pm
+
+=head1 COPYRIGHT
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/script/testappforcomparison_server.pl	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/script/testappforcomparison_server.pl	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,159 @@
+#!/usr/bin/env perl
+
+BEGIN {
+    $ENV{CATALYST_ENGINE} ||= 'HTTP';
+    $ENV{CATALYST_SCRIPT_GEN} = 38;
+    require Catalyst::Engine::HTTP;
+}
+
+use strict;
+use warnings;
+use Getopt::Long;
+use Pod::Usage;
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+
+my $debug             = 0;
+my $fork              = 0;
+my $help              = 0;
+my $host              = undef;
+my $port              = $ENV{TESTAPPFORCOMPARISON_PORT} || $ENV{CATALYST_PORT} || 3000;
+my $keepalive         = 0;
+my $restart           = $ENV{TESTAPPFORCOMPARISON_RELOAD} || $ENV{CATALYST_RELOAD} || 0;
+my $background        = 0;
+my $pidfile           = undef;
+
+my $check_interval;
+my $file_regex;
+my $watch_directory;
+my $follow_symlinks;
+
+my @argv = @ARGV;
+
+GetOptions(
+    'debug|d'             => \$debug,
+    'fork|f'              => \$fork,
+    'help|?'              => \$help,
+    'host=s'              => \$host,
+    'port|p=s'            => \$port,
+    'keepalive|k'         => \$keepalive,
+    'restart|r'           => \$restart,
+    'restartdelay|rd=s'   => \$check_interval,
+    'restartregex|rr=s'   => \$file_regex,
+    'restartdirectory=s@' => \$watch_directory,
+    'followsymlinks'      => \$follow_symlinks,
+    'background'          => \$background,
+    'pidfile=s'           => \$pidfile,
+);
+
+pod2usage(1) if $help;
+
+if ( $debug ) {
+    $ENV{CATALYST_DEBUG} = 1;
+}
+
+# If we load this here, then in the case of a restarter, it does not
+# need to be reloaded for each restart.
+require Catalyst;
+
+# If this isn't done, then the Catalyst::Devel tests for the restarter
+# fail.
+$| = 1 if $ENV{HARNESS_ACTIVE};
+
+my $runner = sub {
+    # This is require instead of use so that the above environment
+    # variables can be set at runtime.
+    require TestAppForComparison;
+
+    TestAppForComparison->run(
+        $port, $host,
+        {
+            argv       => \@argv,
+            'fork'     => $fork,
+            keepalive  => $keepalive,
+            background => $background,
+            pidfile    => $pidfile,
+        }
+    );
+};
+
+if ( $restart ) {
+    die "Cannot run in the background and also watch for changed files.\n"
+        if $background;
+
+    require Catalyst::Restarter;
+
+    my $subclass = Catalyst::Restarter->pick_subclass;
+
+    my %args;
+    $args{follow_symlinks} = 1
+        if $follow_symlinks;
+    $args{directories} = $watch_directory
+        if defined $watch_directory;
+    $args{sleep_interval} = $check_interval
+        if defined $check_interval;
+    $args{filter} = qr/$file_regex/
+        if defined $file_regex;
+
+    my $restarter = $subclass->new(
+        %args,
+        start_sub => $runner,
+    );
+
+    $restarter->run_and_watch;
+}
+else {
+    $runner->();
+}
+
+1;
+
+=head1 NAME
+
+testappforcomparison_server.pl - Catalyst Testserver
+
+=head1 SYNOPSIS
+
+testappforcomparison_server.pl [options]
+
+ Options:
+   -d -debug          force debug mode
+   -f -fork           handle each request in a new process
+                      (defaults to false)
+   -? -help           display this help and exits
+      -host           host (defaults to all)
+   -p -port           port (defaults to 3000)
+   -k -keepalive      enable keep-alive connections
+   -r -restart        restart when files get modified
+                      (defaults to false)
+   -rd -restartdelay  delay between file checks
+                      (ignored if you have Linux::Inotify2 installed)
+   -rr -restartregex  regex match files that trigger
+                      a restart when modified
+                      (defaults to '\.yml$|\.yaml$|\.conf|\.pm$')
+   -restartdirectory  the directory to search for
+                      modified files, can be set mulitple times
+                      (defaults to '[SCRIPT_DIR]/..')
+   -follow_symlinks   follow symlinks in search directories
+                      (defaults to false. this is a no-op on Win32)
+   -background        run the process in the background
+   -pidfile           specify filename for pid file
+
+ See also:
+   perldoc Catalyst::Manual
+   perldoc Catalyst::Manual::Intro
+
+=head1 DESCRIPTION
+
+Run a Catalyst Testserver for this application.
+
+=head1 AUTHORS
+
+Catalyst Contributors, see Catalyst.pm
+
+=head1 COPYRIGHT
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/script/testappforcomparison_test.pl	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/script/testappforcomparison_test.pl	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,53 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Getopt::Long;
+use Pod::Usage;
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use Catalyst::Test 'TestAppForComparison';
+
+my $help = 0;
+
+GetOptions( 'help|?' => \$help );
+
+pod2usage(1) if ( $help || !$ARGV[0] );
+
+print request($ARGV[0])->content . "\n";
+
+1;
+
+=head1 NAME
+
+testappforcomparison_test.pl - Catalyst Test
+
+=head1 SYNOPSIS
+
+testappforcomparison_test.pl [options] uri
+
+ Options:
+   -help    display this help and exits
+
+ Examples:
+   testappforcomparison_test.pl http://localhost/some_action
+   testappforcomparison_test.pl /some_action
+
+ See also:
+   perldoc Catalyst::Manual
+   perldoc Catalyst::Manual::Intro
+
+=head1 DESCRIPTION
+
+Run a Catalyst action from the command line.
+
+=head1 AUTHORS
+
+Catalyst Contributors, see Catalyst.pm
+
+=head1 COPYRIGHT
+
+This library is free software. You can redistribute it and/or modify
+it under the same terms as Perl itself.
+
+=cut
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/t/01app.t	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/t/01app.t	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,8 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use Test::More tests => 2;
+
+BEGIN { use_ok 'Catalyst::Test', 'TestAppForComparison' }
+my $req = request('/');
+ok( $req->is_success, "Request succeeds");
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/t/02pod.t	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/t/02pod.t	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,10 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use Test::More;
+
+eval "use Test::Pod 1.14";
+plan skip_all => 'Test::Pod 1.14 required' if $@;
+plan skip_all => 'set TEST_POD to enable this test' unless $ENV{TEST_POD};
+
+all_pod_files_ok();
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/t/03podcoverage.t	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/t/03podcoverage.t	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,10 @@
+#!/usr/bin/env perl
+use strict;
+use warnings;
+use Test::More;
+
+eval "use Test::Pod::Coverage 1.04";
+plan skip_all => 'Test::Pod::Coverage 1.04 required' if $@;
+plan skip_all => 'set TEST_POD to enable this test' unless $ENV{TEST_POD};
+
+all_pod_coverage_ok();
--- /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/TestAppForComparison/testappforcomparison.conf	1969-12-31 17:00:00.000000000 -0700
+++ /home/devin/web-devel/Catalyst-Devel/1.00/branches/helper_refactor/t/../TestAppForComparison/testappforcomparison.conf	2009-06-15 15:59:41.000000000 -0600
@@ -0,0 +1,3 @@
+# rename this file to TestAppForComparison.yml and put a ':' in front of 'name' if
+# you want to use YAML like in old versions of Catalyst
+name TestAppForComparison
